code/script.js

```diff
@@ -218,108 +221,159 @@ const generateBoard = () => {
+      alert(
+        `Yes, the person has ${value} hair! Keep all people with ${value} hair`
+      );
+      charactersInPlay = charactersInPlay.filter(
+        (person) => person[category] === value
+      );
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      alert(
+        `No, the person doesn't have ${value} hair! Remove all people with ${value} hair`
+      );
+      charactersInPlay = charactersInPlay.filter(
+        (person) => person[category] !== value
+      );
     }
   }
-
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
-  // Invoke a function to redraw the board with the remaining people.
-}
+  // Invoke function to redraw the board with the remaining people.
+  generateBoard();
+};
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
+// If the player wants to guess, the checkMyGuess function invokes.
 const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
-}
+  const playerInteraction = window.confirm("Are you sure you want to guess?");
+
+  if (playerInteraction) {
+    checkMyGuess(personToConfirm);
+  }
+};
 
-// If you confirm, this function is invoked
+//Function for putting in different sound effect to use for winOrLose section
+const playSoundEffect = (audiofile) => {
+  const audio = new Audio(audiofile);
+  audio.play();
+};
+
+// If you confirm previouse guess-function, this function is invoked where different text and sound displays depending on if user wins or loses.
 const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
-}
+  if (personToCheck === secret.name) {
+    alert("YAY, you win!");
+    winOrLoseText.innerHTML = `<h1>HORRAY 🥳</h1> 
+    <h4>Are you on a winning streak?</h4>`;
+    playSoundEffect("assets/win-sound-effect.mp3");
+  } else {
+    alert("Nooooooooooooo....");
+    winOrLoseText.innerHTML = `<h1>"Well" done...🙄</h1> 
+    <h4>Dare to play again?</h4>`;
+    playSoundEffect("assets/lose-sound-effect.mp3");
+  }
+  board.innerHTML = "";
+  winOrLose.style.display = "block";
```

Comment:
Very nice that you made two differens sections for win and lose with fun text, very game-feeling 😊